### DISCUSSION TITLE
‘Drive Details’ and 'Visited' have different map API

### DISCUSSION THREAD

**Discussion by lxl1606** [teslamate-org/teslamate#4329] [General]
‘Drive Details’  use open street map. 'Visited' use Tiles.
how to change OpenStreetMap to Tiles?

**Comment by DrMichael** [comment]
You could change the Layer type of Basemap layer to ArcGIS MapServer...

Interesting discussion though, which looks better... I would prefer Open Street Map in both. As all others do.

**Reply by cwanja** [reply]
Wonder if there's also a limitation or performance with the data point mapping on the visitor dashboard for OpenStreetMaps.

**Reply by DrMichael** [reply]
Visited on RPI3B+ for five years takes some minutes... Do you think the Layer type makes a difference?

**Reply by cwanja** [reply]
Not sure. @swiffer @jheredianet do either of you have thoughts? Is there an index here that could increase the performance? Or could just be rendering challenges, which would be a bakeoff between ArcGIS and OpenStreetMaps.

**Reply by swiffer** [reply]
@DrMichael you are on v1.31.1 ? this change should already have improvement load time:

https://github.com/teslamate-org/teslamate/commit/4d348a1f63e932242a3a6997cf1a8d05d6813262

map rendering is done within browser - could you verify if rendering the maps takes minutes or actual query execution does?

**Reply by cwanja** [reply]
For me, the query takes about 30 - 45 seconds for a **two year** span on a Raspberry Pi 3b+. Select _last 2 years_, wait 30 - 45 seconds for map to render query.

**Reply by DrMichael** [reply]
Actually on 1.31.1-dev.

Time for the query for 5 years is 112004.164 ms (01:52.004). Dashboard with OSM takes indeed ages instead of minutes (just my feeling).

How can I get the time for loading a dashboard/panel?

**Reply by swiffer** [reply]
@DrMichael try this one #4338

**Reply by DrMichael** [reply]
Ok, that needs ~1:30 minutes for dashboard loading. The query itself needs 0:54. With OSM.
With latest also ~1:30 loading of dashboard.

**Reply by swiffer** [reply]
what happens when you modify the query and change `date_trunc('minute', ...)` to `date_trunc('hour', ...)` within the query?

i might introduce 2-3 parameters similar to what i did with the charge level dashboard

**Reply by DrMichael** [reply]
Interesting, I increased the swap file from 100 to 1024. Loading time went down to ~50 seconds with latest.

Running #4338, loading time with minutes also ~50 seconds. With hours appr. also (and it does not look good).

**Reply by swiffer** [reply]
I just don't get why rendering the map takes longer than the query exec time alone - it's nearly instant here.

**Reply by DrMichael** [reply]
Hmmm, why do you think so?

With edge I have ~50 seconds for dashboard loading (which should include the rendering and query time) and 43 seconds, when I run the query with psql in the container. (With 1024 swap file...)

**Reply by swiffer** [reply]
because of what you've written here:

> Ok, that needs ~1:30 minutes for dashboard loading. The query itself needs 0:54. With OSM.

**Reply by DrMichael** [reply]
Ah, that. Apparently caused by too small swap file. After inceasing it, is is 50 seconds versus 43.
